src/posix/platform/rcp_caps_diag.cpp

```diff
@@ -498,22 +502,149 @@ void RcpCapsDiag::TestSpinelCommands(Category aCategory)
+    {
+        if (entry.mCategory != aCategory)
+        {
+            continue;
+        }
+
+        OutputFormat(RadioCapbilityToString(entry.mFlag), SupportToString((aRadioCapFlags & entry.mFlag) > 0));
+    }
+}
+
+void RcpCapsDiag::TestSpinelCapbilityFlags(void)
+{
+    static constexpr uint8_t kCapsBufferSize = 100;
+    otError                  error;
+    uint8_t                  capsBuffer[kCapsBufferSize];
+    spinel_size_t            capsLength = sizeof(capsBuffer);
+
+    SuccessOrExit(error = mRadioSpinel.Get(SPINEL_PROP_CAPS, SPINEL_DATATYPE_DATA_S, capsBuffer, &capsLength));
+
+    Output("\r\nSpinel Capbility Flags :\r\n");
+    TestSpinelCapbilityFlags(kCategoryBasic, capsBuffer, capsLength);
+    TestSpinelCapbilityFlags(kCategoryUtils, capsBuffer, capsLength);
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        Output("Failed to get Spinel capbility flags: %s", otThreadErrorToString(error));
+    }
+
+    return;
+}
+
+void RcpCapsDiag::TestSpinelCapbilityFlags(Category aCategory, const uint8_t *aCapsData, spinel_size_t aCapsLength)
+{
+    static constexpr uint8_t   kCapsNameSize        = 40;
+    static constexpr FlagEntry kSpinelFlagEntries[] = {{kCategoryBasic, SPINEL_CAP_CONFIG_RADIO},
+                                                       {kCategoryBasic, SPINEL_CAP_MAC_RAW},
+                                                       {kCategoryBasic, SPINEL_CAP_RCP_API_VERSION},
+                                                       {kCategoryUtils, SPINEL_CAP_OPENTHREAD_LOG_METADATA},
+                                                       {kCategoryUtils, SPINEL_CAP_RCP_MIN_HOST_API_VERSION},
+                                                       {kCategoryUtils, SPINEL_CAP_RCP_RESET_TO_BOOTLOADER}};
+    char                       capName[kCapsNameSize];
+
+    Output("\r\n%s :\r\n", CategoryToString(aCategory));
+
+    for (const FlagEntry &entry : kSpinelFlagEntries)
+    {
+        if (entry.mCategory != aCategory)
+        {
+            continue;
+        }
+
+        snprintf(capName, sizeof(capName), "SPINEL_CAPS_%s", spinel_capability_to_cstr(entry.mFlag));
+        OutputFormat(capName, SupportToString(IsSpinelCapabilitySupported(aCapsData, aCapsLength, entry.mFlag)));
+    }
+}
+
+bool RcpCapsDiag::IsSpinelCapabilitySupported(const uint8_t *aCapsData, spinel_size_t aCapsLength, uint32_t aCapability)
+{
+    bool ret = false;
+
+    while (aCapsLength > 0)
+    {
+        unsigned int   capability;
+        spinel_ssize_t unpacked;
+
+        unpacked = spinel_datatype_unpack(aCapsData, aCapsLength, SPINEL_DATATYPE_UINT_PACKED_S, &capability);
+        VerifyOrExit(unpacked > 0);
+        VerifyOrExit(capability != aCapability, ret = true);
+
+        aCapsData += unpacked;
+        aCapsLength -= static_cast<spinel_size_t>(unpacked);
+    }
+
+exit:
+    return ret;
+}
+
+void RcpCapsDiag::OutputFormat(const char *aName, const char *aValue)
+{
+    static constexpr uint8_t kMaxLength    = 56;
```

Comment:
Should this be equal to the `kMaxLength` defined in `OutputResult`? If so, what about moving these constexpr definition into the class private section?

Reply:
They are two different values. I have renamed these two values to make the definition clearer.